src/hooks/useExpenditureStaking.ts

```diff
@@ -0,0 +1,82 @@
+import { Extension, Id } from '@colony/colony-js';
+import { BigNumber } from 'ethers';
+import { useEffect, useState } from 'react';
+
+import { ADDRESS_ZERO } from '~constants';
+import { useAppContext } from '~context/AppContext.tsx';
+import { useColonyContext } from '~context/ColonyContext.tsx';
+import { useGetUserReputationQuery } from '~gql';
+import useEnoughTokensForStaking from '~hooks/useEnoughTokensForStaking.ts';
+import useExtensionData from '~hooks/useExtensionData.ts';
+import { isInstalledExtensionData } from '~utils/extensions.ts';
+
+const useExpenditureStaking = (createdInDomainId = Id.RootDomain) => {
+  const { colony } = useColonyContext();
+  const { user } = useAppContext();
+  const { walletAddress = '' } = user ?? {};
+
+  const [stakeAmount, setStakeAmount] = useState<string>();
+
+  const { data, loading: reputationLoading } = useGetUserReputationQuery({
+    fetchPolicy: 'network-only',
+    variables: {
+      input: {
+        colonyAddress: colony.colonyAddress,
+        walletAddress: ADDRESS_ZERO,
```

Comment:
intentional or do we want `walletAddress` here?

Reply:
Super non-obvious but passing `ADDRESS_ZERO` as walletAddress to this query will return reputation for the entire domain. If you passed an address of a particular user, you'd get their reputation amount.

Reply:
makes sense, gotcha :+1: